explorer/utils.py

```diff
@@ -238,9 +237,45 @@ def get_s3_connection():
                              app_settings.S3_SECRET_KEY,
                              default_bucket=app_settings.S3_BUCKET)
 
+
 def compare_sql(old_sql, new_sql):
     """
     Compares whether two sql queries are the 
     same after formatting them
     """
     return fmt_sql(old_sql) == fmt_sql(new_sql)
+
+
+def get_duration(td):
+    """
+    Show time intervals (timedelta) in hours, minutes and seconds
+    """
+    total_seconds = int(td.total_seconds())
+    hours = total_seconds // 3600
+    minutes = (total_seconds % 3600) // 60
+    seconds = total_seconds - (hours * 3600) - (minutes * 60)
+
+    if hours == 0:
+        return '{} minutes and {} seconds'.format(minutes, seconds)
+
+    return '{} hours, {} minutes and {} seconds'.format(hours, minutes, seconds)
```

Comment:
see if there's a better way to humanize timedelta (which is what this is essntially). see under django & dateutil packages

Reply:
I've updated this part - using [ago](https://pypi.org/project/ago/) to humanize timedelta.